src/lib/Scenes/Show/Components/ShowHeader/Components/ArtistNamesList.tsx

```diff
@@ -1,33 +1,49 @@
-import { Sans } from "@artsy/palette"
+import { Flex, Sans } from "@artsy/palette"
+import SwitchBoard from "lib/NativeModules/SwitchBoard"
 import React from "react"
+import { TouchableOpacity } from "react-native"
 
-export const ArtistNamesList = ({ artists }) => {
+export const ArtistNamesList = ({ artists, component, viewAllArtists }) => {
   if (!artists || !artists.length) {
     return null
   }
-
+  const urlMappedToArtist = {}
   const names = artists
     .map(artist => artist.name)
     .slice(0, 5)
     .join(", ")
+
+  artists.map(artist => (urlMappedToArtist[artist.name] = artist.href)).slice(0, 5)
```

Comment:
```js
const urlMappedToArtist = artists.reduce((artistUrls, artist) => {
  return {
    ...artistUrls,
    [artist.name]: artist.href
  }
}, {})
```

[reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) is one of the most powerful concepts in javascript once its understood, because it basically allows you to take one type of object and transform it into anything else. Its handy 😄

Reply:
Thanks Chris! Great suggestions! I ended up not needed to use reduce in the refactor but its definitely a clean method I should be using more often!